icu4c/source/common/dictbe.cpp

```diff
@@ -1381,9 +1432,33 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     }
     (void)prevCPPos; // suppress compiler warnings about unused variable
 
+    UChar32 lastChar = utext_char32At(inText,rangeEnd);
+    if (!foundBreaks.isEmpty()
+            && foundBreaks.peeki() == rangeEnd
+            && !isOpenPunct(lastChar)
+            && !fNumberWordSet.contains(lastChar)) {
+        foundBreaks.popi();
+        correctedNumBreaks--;
+    }
+
     // inString goes out of scope
     // inputMap goes out of scope
-    return numBreaks;
+    return correctedNumBreaks;
+}
+
+void CjkBreakEngine::loadJpParticleAndAuxVerbs() {
+    UErrorCode error = U_ZERO_ERROR;
```

Comment:
The error code should be pass in from the constructor by reference instead adding a new one in the constructor and got ignored later. If any error happen, the caller of the constructor should get the error.